tests/test_lifespan.py

```diff
@@ -158,13 +160,17 @@ async def test():
 @pytest.mark.parametrize("mode", ("auto", "on"))
 def test_lifespan_scope_asgi3app(mode):
     async def asgi3app(scope, receive, send):
-        assert scope == {"version": "3.0", "spec_version": "2.0"}
+        assert scope == {
+            "asgi": {"spec_version": "2.0", "version": "3.0"},
+            "type": "lifespan",
+        }
 
     async def test():
         config = Config(app=asgi3app, lifespan=mode)
         lifespan = LifespanOn(config)
 
         await lifespan.startup()
+        await lifespan.main()
```

Comment:
Hmm, this feels weird. `lifespan.startup()` already schedules `lifespain.main()` on the event loop:

https://github.com/encode/uvicorn/blob/a9a46e87d8ba41f08b3a401b7b0cc572333a7193/uvicorn/lifespan/on.py#L29

I think the core problem with the lifespan implementation here is that we fire off this task, but never await it, so exceptions are never bubbled up.

Reply:
Filed #913 to illustrate. :)

Reply:
yes this is way better I missed that error_occured !